plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -158,20 +166,32 @@ func (k *KafkaConsumer) Init() error {
 		config.Version = version
 	}
 
+	if k.EnableTLS {
+		config.Net.TLS.Enable = true
+	}
+
 	tlsConfig, err := k.ClientConfig.TLSConfig()
 	if err != nil {
 		return err
 	}
 
 	if tlsConfig != nil {
 		config.Net.TLS.Config = tlsConfig
+
+		// TLS is forced on if a custom tls.Config is set.
 		config.Net.TLS.Enable = true
```

Comment:
If my config had custom TLS settings and also "enable_tls = false" I would expect the TLS config to be ignored and TLS not to be used.  This line would enable tls even though I explicitly said not to.

Reply:
I'll update the option to be a `*bool` so we can have the distinction between unset and `enable_tls = false`.